src/ghost-content/index.js

```diff
@@ -141,21 +143,71 @@ module.exports = ({ logger, db, projectLocation, enabled }) => {
 
   const updatePendingForAllFolders = () => Promise.each(trackedFolders, updatePendingForFolder)
 
-  const recordRevision = async (rootFolder, file, content) => {
+  const recordRevision = async (rootFolder, file, newContent) => {
     const knex = await db.get()
 
     const { normalizedFolderName } = normalizeFolder(rootFolder)
 
-    const id = await knex('ghost_content')
+    const { id, content } =
+      (await knex('ghost_content')
+        .where({ folder: normalizedFolderName, file })
+        .select('id', 'content')
+        .get(0)) || {}
+
+    if (newContent === content) {
+      return Promise.resolve()
+    }
+
+    const revision = uuid.v4()
+
+    return knex.transaction(trx => {
+      upsert({
+        knex,
+        tableName: 'ghost_content',
+        where: { folder: normalizedFolderName, file },
+        data: { content: newContent },
+        transaction: trx
+      })
+        .then(() =>
+          knex('ghost_revisions')
+            .transacting(trx)
+            .insert({
+              content_id: id,
+              revision,
+              created_by: 'admin'
+            })
+        )
+        .then(trx.commit)
+        .then(() => updatePendingForFolder(normalizedFolderName))
+        .catch(err => {
+          logger.error('[Ghost Content Manager]', err)
+          trx.rollback()
+        })
+    })
+  }
+
+  const readFile = async (rootFolder, file) => {
+    const knex = await db.get()
+    const { normalizedFolderName } = normalizeFolder(rootFolder)
+    return knex('ghost_content')
+      .select('content')
       .where({ folder: normalizedFolderName, file })
-      .select('id')
       .get(0)
-      .get('id')
+      .get('content')
+  }
+
+  const deleteFile = async (rootFolder, file) => {
+    const knex = await db.get()
+    const { normalizedFolderName } = normalizeFolder(rootFolder)
+
+    const { id } =
+      (await knex('ghost_content')
+        .where({ folder: normalizedFolderName, file, deleted: false })
+        .select('id')
+        .get(0)) || {}
 
     if (!id) {
-      throw new Error(
-        `No Ghost content for file: ${file} in folder: ${normalizedFolderName}. Cannot record the new revision.`
-      )
+      throw new Error(`Can't delete file: ${file}: couldn't find it in folder: ${normalizedFolderName}`)
     }
 
     const revision = uuid.v4()
```

Comment:
This duplicates the code for `recordRevision`. Not critical but could go with a helper private method `_recordRevison(contentId)` that will keep the revision number generation and insertion in the revisions table

Reply:
Good point, moved it to separate method and renamed `recordRevision` to `upsertFile`.